test/e2e/connectivity_test.go

```diff
@@ -136,34 +146,54 @@ func TestHostPortPodConnectivity(t *testing.T) {
 }
 
-// createPodsOnDifferentNodes creates numPods busybox test Pods and assign them to all the different
-// Nodes in round-robin fashion, then returns the names of the created Pods as well as a function
-// which will delete the Pods when called.
-func createPodsOnDifferentNodes(t *testing.T, data *TestData, numPods int) (podNames []string, cleanup func()) {
-	podNames = make([]string, 0, numPods)
+// createPodsOnDifferentNodes creates agnhost Pods through a DaemonSet. This function returns information of the created
+// Pods as well as a function which will delete the Pods when called.
+func createPodsOnDifferentNodes(t *testing.T, data *TestData) (podInfos []podInfo, cleanup func() error) {
+	dsName := "connectivity-test"
+	_, cleanup, err := data.createDaemonSet(dsName, testNamespace, agnhostContainerName, agnhostImage, []string{"sleep", "3600"}, nil)
+	if err != nil {
+		t.Fatalf("Error when creating DaemonSet '%s': %v", dsName, err)
+	}
+	if err := data.waitForDaemonSetPods(defaultTimeout, dsName, testNamespace); err != nil {
+		t.Fatalf("Error when waiting for DaemonSet Pods to get IPs: %v", err)
+	}
 
-	cleanup = func() {
-		for _, podName := range podNames {
-			deletePodWrapper(t, data, podName)
-		}
+	piMap := map[string][]podInfo{"linux": {}, "windows": {}}
+	pods, err := data.clientset.CoreV1().Pods(testNamespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "antrea-e2e=" + dsName,
+	})
+	if err != nil {
+		t.Fatalf("Error when getting connectivity test Pods: %v", err)
 	}
 
-	for idx := 0; idx < numPods; idx++ {
-		podName := randName(fmt.Sprintf("test-pod-%d-", idx))
-		nodeName := nodeName(idx % clusterInfo.numNodes)
-		t.Logf("Creating busybox test Pods '%s' on '%s'", podName, nodeName)
-		if err := data.createBusyboxPodOnNode(podName, nodeName); err != nil {
-			cleanup()
-			t.Fatalf("Error when creating busybox test Pod '%s': %v", podName, err)
-		}
-		podNames = append(podNames, podName)
+	for _, p := range pods.Items {
+		os := clusterInfo.nodesOS[p.Spec.NodeName]
+		piMap[os] = append(piMap[os], podInfo{p.Name, os})
+	}
+	var linIdx, winIdx int
+	for linIdx != len(piMap["linux"]) && winIdx != len(piMap["windows"]) {
+		podInfos = append(podInfos, piMap["linux"][linIdx])
+		podInfos = append(podInfos, piMap["windows"][winIdx])
+		linIdx++
+		winIdx++
+	}
+	for ; linIdx != len(piMap["linux"]); linIdx++ {
+		podInfos = append(podInfos, piMap["linux"][linIdx])
+	}
+	for ; winIdx != len(piMap["windows"]); winIdx++ {
+		podInfos = append(podInfos, piMap["windows"][winIdx])
 	}
 
-	return podNames, cleanup
+	return podInfos, cleanup
 }
 
 func (data *TestData) testPodConnectivityDifferentNodes(t *testing.T) {
 	numPods := 2
+	// If there're Windows Nodes, an ideal testbed is a cluster of 1 Control Plane Node, 1 Linux worker and 2
+	// Windows Nodes, which is the setup of Windows CI testbed.
+	if len(clusterInfo.windowsNodes) > 1 && (clusterInfo.numNodes-len(clusterInfo.windowsNodes)) > 1 {
+		numPods = 3
+	}
```

Comment:
what do you think of just:
```go
maxPods := 3
if len(podInfos) > maxPods {
    podInfos = podsInfos[:maxPods]
}
```
I don't think we need to handle the Windows case in any special way here.

Reply:
Yes, we can update like this. But we need to add an if after that like:
```
	if numPods > maxPods {
		numPods = maxPods
	}
```

Without it, if the cluster is of 5 Linux nodes and Hybrid mode, then numPods is 5, podInfos slice length is 3. `podInfos[:numPods]` will return error.

Reply:
I think we should just set `maxPods` to `clusterInfo.numNodes` for the hybrid case